lib/ssl/src/tls_connection.erl

```diff
@@ -669,21 +670,24 @@ hello(internal, #client_hello{extensions = Extensions} = Hello,
                                          handshake_env = HsEnv#handshake_env{hello = Hello}},
      [{reply, From, {ok, Extensions}}]};
 hello(internal, #server_hello{extensions = Extensions} = Hello,
-      #state{ssl_options = #{handshake := hello},
+      #state{ssl_options = #{
+                 handshake := hello,
+                 ocsp_stapling := OcspStapling},
              handshake_env = HsEnv,
              start_or_recv_from = From} = State) ->
     %% RFC6066.8, If a server returns a "CertificateStatus" message,
     %% then the server MUST have included an extension of type
     %% "status_request" with empty "extension_data" in the extended
     %% server hello.
     OcspState = HsEnv#handshake_env.ocsp_stapling_state,
-    IsCertStatusRequested = maps:is_key(ocsp_stapling, Extensions),
+    IsCertStatusRequested = (undefined == maps:get(status_request, Extensions, false)),
     {next_state, user_hello,
      State#state{start_or_recv_from = undefined,
                  handshake_env = HsEnv#handshake_env{
                      hello = Hello,
                      ocsp_stapling_state = OcspState#{
-                         is_server_hello_requested := IsCertStatusRequested}}},
+                         ocsp_negotiated :=
```

Comment:
Following the above logic this can be simplified to
```erlang
ocsp_negotiated := OcspNegotiated,
```

Reply:
done